fdbserver/ConfigNode.actor.cpp

```diff
@@ -316,9 +316,13 @@ class ConfigNodeImpl {
 
 	ACTOR static Future<Void> commit(ConfigNodeImpl* self, ConfigTransactionCommitRequest req) {
 		ConfigGeneration currentGeneration = wait(getGeneration(self));
-		if (req.generation != currentGeneration) {
+		if (req.generation.committedVersion != currentGeneration.committedVersion) {
 			++self->failedCommits;
-			req.reply.sendError(transaction_too_old());
+			req.reply.sendError(commit_unknown_result());
```

Comment:
I think my confusion stems from the fact that in this scenario, C sending `commit_unknown_result` actually signifies C responding with "I'm not rejecting this commit but I also can't accept it because my version is too old" which to me seems a little different than what `commit_unknown_result` typically means. But I understand why sending `not_committed` is the incorrect thing to do based on how the counters in `PaxosConfigTransaction` are set up.

Reply:
Yeah, it might make sense to create a new error code for this scenario. `C` sending `commit_unknown_result` doesn't mean this will actually be thrown from the client, just that it will cause `CommitQuorum::maybeCommitted` to increment.